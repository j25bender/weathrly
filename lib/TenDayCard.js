// import React, { Component } from 'react';
// import '../styles/TenDayCard.css';

// export default class TenDayCard extends React.Component {
//     constructor() {
//         super();        
//     }

//     render () {
//         return (
//             <div className="tenDayCard">
//                 <div>{ this.props.data.day }</div>
//                 <div>{ this.props.data.low }˚F / { this.props.data.high }˚F</div>
//                 <img className="weather-icon" src={ this.props.data.icon_url } />
//             </div>
//         )
//     }
// }
