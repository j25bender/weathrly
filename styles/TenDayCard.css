.cards {
    background: #0f1926;
    margin: 10px auto 0px;
    color: #fff;    
    min-width: 5.5rem;
    padding: 0;
    padding-bottom: 25px;    
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: center;
    width: 90%;
}

.tenDayCard {
  background: #2a3c57;
  margin: 1rem;
  max-width: 15rem;
  min-width: 5.5rem;
  padding: 1rem;
}

.tenDayHeader h3 {
  margin-bottom: 20px;  
}
  
.temp {
  margin: 0;
}

.weather-icon {
    width: 40%;
}

.tenDayHeader {
  margin-top: 8px;
  font-size: 1.8rem;
  padding-left: 2rem;
  text-align: left;
  width: 100%;
}
  
@media screen and (max-width: 1010px) {
  .card {
    align-content: flex-start;
    flex-wrap: wrap;
    min-width: 15rem;
  }

  .tenDayHeader {
      text-align: center;
  }
}