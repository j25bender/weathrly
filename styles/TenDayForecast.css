.card {
    flex: 1;
    border: 0.01rem solid ghostwhite;
    margin: 1rem;
    max-width: 15rem;
    min-width: 5.5rem;
    padding: 1rem;
}