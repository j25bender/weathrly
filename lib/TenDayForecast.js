import React from 'react';
import TenDayCard from './TenDayCard.js';

function TenDayForecast (props) {
  return (
    <div className="cards">
        <div className="tenDayHeader">
            <h3>10 Day Forecast</h3>
        </div>
        <div className="card">
        {
            props.forecast.map( (card, index) => <TenDayCard key={index} data={card} /> )
        }
        </div>
    </div>
  )
}

export default TenDayForecast;