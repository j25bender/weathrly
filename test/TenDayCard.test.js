import React from 'react';
import { shallow, mount } from 'enzyme';
import TenDayForecast from '../lib/TenDayForecast';
import TenDayCard from '../lib/TenDayCard';
import cleanData from '../lib/cleanData';
import mockData from '../lib/mockData';
import 'jest-localstorage-mock';

describe('TenDayForecast Tests ', () => {
    let filteredData;
    let wrapper;
    
    beforeEach(() => {
        filteredData = cleanData(mockData);
        wrapper = mount(<TenDayForecast forecast={filteredData.tenDayArray}/>);
    });

    it('TenDayForecast component should exist', () => {
        expect(wrapper).toBeDefined()
        expect(wrapper.find('h3').text()).toEqual('10 Day Forecast')        
    });

    it('TenDayForecast should be a function', () => {
        expect(typeof TenDayForecast).toEqual('function');
    });

    it('TenDayForecast should render 10 TenDayCard components', () => {
        expect(wrapper.find('TenDayCard').length).toEqual(10);
        expect(TenDayCard).toMatchSnapshot();
    });
  
    it('Each TenDayCard component should render a day, high / low temp and an icon via an image url', () => {
        const firstCard = wrapper.find('TenDayCard').first()
        const lastCard = wrapper.find('TenDayCard').last()

        expect(firstCard.props().data.day).toEqual('Wednesday')
        expect(lastCard.props().data.day).toEqual('Friday')

        expect(firstCard.props().data.high).toEqual('51')
        expect(lastCard.props().data.high).toEqual('31')

        expect(firstCard.props().data.low).toEqual('32')
        expect(lastCard.props().data.low).toEqual('18')

        expect(wrapper.find('.weather-icon').length).toEqual(10);
        expect(firstCard.props().data.icon_url).toEqual('http://icons.wxug.com/i/c/k/partlycloudy.gif')
        expect(lastCard.props().data.icon_url).toEqual('http://icons.wxug.com/i/c/k/partlycloudy.gif')
    });
});