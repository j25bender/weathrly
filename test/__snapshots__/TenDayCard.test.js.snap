// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TenDayForecast Tests  TenDayForecast should render 10 TenDayCard components 1`] = `[Function]`;
